Dev

sergei-gots opened the pull request:
Pet-Shelter-Bot working prototype.
- bot handles commands getting from user  and offers user tree-like buttons menu
- bot shows information about shelter and adopting details
- bot is able to retrieve from adopter their contacts and report about a pet on trial 
- bot performs dialog for the user with a shelter volunteer
- API for Front-End provided

ToDo: 
- handle photo (is being developed); 
- notifications (timer)
- tests

sergei-gots committed:
processMessage reduced

vzavolsky committed:
Merge pull request #4 from Sergei-Gots/feature-S-01

Feature s 01

sergei-gots committed:
Cats dev stage handler

sergei-gots committed:
Update CatsDevStageHandler.java

sergei-gots committed:
Update CatsDevStageHandler.java : text.startsWith("/dev-test-")

ATsarev01 committed:
Add ShelterHandler.class to make a choice of shelters

ATsarev01 committed:
Create StartHandler and add comments

ATsarev01 committed:
Add documentation

ATsarev01 committed:
Update documentation

ATsarev01 committed:
Merge branch 'dev' into feature-A-02

ATsarev01 committed:
Merge pull request #5 from Sergei-Gots/feature-A-02

Add ShelterHandler to make a choice of shelters

ATsarev01 committed:
Update CatsDevStageHandler.java

sergei-gots committed:
column cat.adoption_date added

sergei-gots committed:
column cat.adoption_date added

sergei-gots committed:
@Entity Pet and table 'pets' added  instead of Cat/'cats'

ATsarev01 committed:
Add ShelterInfoHandler

ATsarev01 committed:
Add ShelterInfoHandler and delete unusable method

ATsarev01 committed:
Delete buttons after pressing by user

sergei-gots committed:
PetRepository introduced instead of CatRepository

sergei-gots committed:
@Entity Shelter/ table 'shelters' added

sergei-gots committed:
AbstractAdopter removed

ATsarev01 committed:
Add schedule_info in ShelterInfoHandler

sergei-gots committed:
PetService.createCat/Dog() methods added

sergei-gots committed:
Update JavaDoc in DevStageDBHandler.java

ATsarev01 committed:
Update TelegramBotListener.java

Edit file

ATsarev01 committed:
Separate dog & cat handlers

ATsarev01 committed:
Merge pull request #8 from java-dev-int-4-team-1/feature-A-03

Feature a 03

ATsarev01 committed:
Merge pull request #9 from java-dev-int-4-team-1/feature-A-04

Separate dog & cat handlers

sergei-gots committed:
Merge pull request #7 from java-dev-int-4-team-1/feature-S-02-db-entities-correction

Feature s 02 DB entities correction

vzavolsky committed:
v01

vzavolsky committed:
UserMessages and Buttons

vzavolsky committed:
Merge pull request #11 from java-dev-int-4-team-1/feature-V-01

Feature v 01

vzavolsky committed:
//test

vzavolsky committed:
Оптимизация классов Handler и Entity.
Основная логика выборки сообщиений и генерации кнопок из таблиц.

vzavolsky committed:
UserMessageRepository - поправил имя метода.

vzavolsky committed:
application.properties - чистка от персональных данных.

vzavolsky committed:
Merge pull request #13 from java-dev-int-4-team-1/feature-V-20230707

Выборка данных из таблиц

sergei-gots committed:
Migrations for 'shelters', 'buttons' added

vzavolsky committed:
Merge pull request #15 from java-dev-int-4-team-1/feature-S-20230709-migrations

Migrations for 'shelters', 'buttons' added

sergei-gots committed:
user-messages - mirgrations added

vzavolsky committed:
Merge pull request #17 from java-dev-int-4-team-1/feature-S-20230709-migrations

user-messages - mirgrations added

sergei-gots committed:
Volunteer-call Initial commit: Listener and ShelterHandler reduced to canonical form

sergei-gots committed:
Volunteer-call request call scheme added

sergei-gots committed:
Volunteer-call prototype created

sergei-gots committed:
@Entity Volunteer added; @Entity Pet fixed

sergei-gots committed:
user_messages.shelter_id as FK

sergei-gots committed:
ShelterHandler, StartHandler ButtonRepository оптимизированы

sergei-gots committed:
ShelterHandler.makeButtonList optimized

sergei-gots committed:
@Entities Volunteer, Dialog added. VolunteerHandler is in progress

sergei-gots committed:
Чат с волонтёром умеет начинаться

sergei-gots committed:
Чат с волонтёром умеет начинаться

sergei-gots committed:
Чат с волонтёром принимает команду /join

sergei-gots committed:
Fixed - выход из обработчика ShelterHandler.handle (CallbackQuery)
Если VolunteerHandler.handle обработал команду /join

vzavolsky committed:
Merge pull request #19 from java-dev-int-4-team-1/feature-S-20230709-volunteer-call

Volunteer-call

dmitry-bizin commented on src/main/java/pro/sky/petshelterbot/entity/UserMessage.java:
```diff
@@ -0,0 +1,47 @@
+    @Lob
+    @Type(type = "org.hibernate.type.TextType")
+    private String message;
+    private Long shelterId;
```
здесь хорошо бы Shelter использовать и связь проставить

sergei-gots replied:
fixed

dmitry-bizin commented on src/main/java/pro/sky/petshelterbot/entity/Pet.java:
```diff
@@ -0,0 +1,111 @@
+    }
+
+    public Pet(String species, String name, Shelter shelter) {
+        this(-1, species, name, shelter, false);
```
а почему -1? лучше сделать Long тип и null присвоить

sergei-gots replied:
fixed

dmitry-bizin commented on src/main/java/pro/sky/petshelterbot/handler/ShelterInfoHandler.java:
```diff
@@ -0,0 +1,88 @@
+
+    @Override
+    public boolean handle(Message message, String key, Long chatId, Long shelterId) {
+        if("schedule_info".equals(key)) {
```
строки вынести бы в константы, чтобы на них везде ссылаться и не нарваться на опечатку случайную и баг

sergei-gots replied:
да, сталкивался уже с неточностью.

dmitry-bizin commented on src/main/java/pro/sky/petshelterbot/repository/ShelterRepository.java:
```diff
@@ -0,0 +1,16 @@
+public interface ShelterRepository extends JpaRepository<Shelter,Long> {
+    Optional<Shelter> findByType(String type);
+
+    default Shelter getCatShelter() { return findByType("cat")
```
тип лучше сделать enum-ом (и в сущности тоже)

sergei-gots replied:
рассмотрим вариант решения ещё раз

sergei-gots committed:
Fix по review Дмитрия + Расширенный Fix по проекту: убраны неточности и добавлены обобщения (AbstractHandler.getShelter(), .getUserMessage())

sergei-gots committed:
Cleanup убраны неактуальные неиспользуемые классы и методы

sergei-gots committed:
Shelter.SHELTER_TYPE CAT, DOG - enum set added to operate shelters within developing code.

sergei-gots committed:
Dialog стал поддерживаться на стороне пользователя. VolunteerChatHandler refactored to DialogHandler

sergei-gots committed:
Merge pull request #21 from java-dev-int-4-team-1/feature-S-20230710-cleanup

Feature s 20230710 clean up

sergei-gots committed:
Merge pull request #22 from java-dev-int-4-team-1/feature-S-20230710-clean-up

enum Shelter.SHELTER_TYPE { CAT, DOG } intorduced

sergei-gots committed:
Диалог волонтёра с адоптером. Базовая версия завершена. Можно покрывать тестами. ToDo v2 - добавить кнопок в подвал и разрешить волонтёру уходить с работы)

sergei-gots committed:
Merge pull request #23 from java-dev-int-4-team-1/feature-S-20230711-dialog-adopter-volunteer

vzavolsky committed:
Shelter и Pet конроллеры.

ATsarev01 committed:
Add ShelterInfoHandlerTest

ATsarev01 committed:
Add ShelterInfoHandlerTest

sergei-gots committed:
Update TelegramBotConfiguration.java

Annotation @PropertySource commented

sergei-gots committed:
Merge pull request #24 from java-dev-int-4-team-1/feature-A-test1

Add prototype ShelterInfoHandlerTest

sergei-gots committed:
Update TelegramBotConfiguration.java

sergei-gots committed:
Merge pull request #24 from java-dev-int-4-team-1/feature-A-test1

Add ShelterInfoHandlerTest

sergei-gots committed:
Update TelegramBotConfiguration.java

vzavolsky committed:
Shelter и Pet (cat, dog) конроллеры.

sergei-gots committed:
table 'reports' added

sergei-gots committed:
table 'reports' added

sergei-gots committed:
Merge pull request #25 from java-dev-int-4-team-1/feature-V-20230711

Shelter, Pet (dog, cat) контроллеры.

sergei-gots committed:
@Entity Report added

sergei-gots committed:
@Entity Report added

sergei-gots committed:
Pet.adopter (pets.adopter_id) added to the @Entity Pet

vzavolsky committed:
Merge branch 'dev' into feature-S-20230711-Report-table-n-entity

vzavolsky committed:
Merge pull request #26 from java-dev-int-4-team-1/feature-S-20230711-Report-table-n-entity

Feature s 20230711 report table n entity

vzavolsky committed:
Report (cat, dog) конроллеры.

vzavolsky committed:
List<Report> (cat, dog) конроллеры.

vzavolsky committed:
List<Report> (cat, dog) конроллеры.

vzavolsky committed:
Merge pull request #27 from java-dev-int-4-team-1/feature-V-20230711

Переименовал классы контроллеров для собак и кошек.

vzavolsky committed:
Чистка конструкторов от id, удаление сеттеров.

vzavolsky committed:
Merge pull request #28 from java-dev-int-4-team-1/feature-V-20230711

Чистка конструкторов: id, setId.

vzavolsky committed:
Report API.

vzavolsky committed:
Report API.

dmitry-bizin commented on src/main/java/pro/sky/petshelterbot/controllers/ShelterController.java:
```diff
@@ -0,0 +1,67 @@
+
+    @PostMapping(path = "/add")
+    public ResponseEntity<Shelter> add(
+            @RequestParam String name,
```
а почему не в теле запроса данные передаёте? лучше в теле передавать

sergei-gots replied:
Здесь  и далее полагаю имеется в виду передача параметра через @PathVariable

dmitry-bizin replied:
через тело запроса - RequestBody

dmitry-bizin replied:
по-прежнему актуально

dmitry-bizin commented on src/main/java/pro/sky/petshelterbot/controllers/ShelterController.java:
```diff
@@ -0,0 +1,67 @@
+        return ResponseEntity.ok(shelterService.add(shelter));
+    }
+
+    @PostMapping(path = "/add")
```
не нужно add, просто слэш

dmitry-bizin replied:
по-прежнему актуально

dmitry-bizin commented on src/main/java/pro/sky/petshelterbot/controllers/ShelterController.java:
```diff
@@ -0,0 +1,67 @@
+        return ResponseEntity.ok(shelterService.delete(id));
+    }
+
+    @GetMapping(path = "/{$id}")
```
зачем доллар?

dmitry-bizin replied:
по-прежнему актуально

dmitry-bizin commented on src/main/java/pro/sky/petshelterbot/controllers/ShelterController.java:
```diff
@@ -0,0 +1,67 @@
+        return ResponseEntity.ok(shelterService.findAll());
+    }
+
+    @DeleteMapping(path = "/{$id}")
```
зачем доллар?

dmitry-bizin replied:
по-прежнему актуально

dmitry-bizin commented on src/main/java/pro/sky/petshelterbot/controllers/ShelterController.java:
```diff
@@ -0,0 +1,67 @@
+        return ResponseEntity.ok(shelterService.get(id));
+    }
+
+    @GetMapping(path = "/update/{id}")
```
почему GetMapping для адпейта? update не нужно, сразу слэш и id

sergei-gots replied:
полагаю, что @PutMapping

а параметр - @RequestBody Shelter

dmitry-bizin replied:
да

dmitry-bizin commented on src/main/java/pro/sky/petshelterbot/controllers/ShelterController.java:
```diff
@@ -0,0 +1,67 @@
+    public ResponseEntity<Shelter> update(
+            @RequestParam Long id,
+            @RequestParam(required = false) String name,
+            @RequestParam(required = false) String workTime,
```
а почему не в теле запроса данные передаёте? лучше в теле передавать

sergei-gots replied:
@RequestBody Shelter т.е.

dmitry-bizin replied:
да

dmitry-bizin replied:
по-прежнему актуально

sergei-gots committed:
/join_dialog, /start - handlers for volunteer dialog  completed

sergei-gots committed:
Dialog business-logic completed

vzavolsky committed:
Контроллеры

@RequestMapping: /cat-shelter, /dog-shelter, /pet, /adopter
 GET /adopter/all-ready-to-adopt
 POST /dog-shelter/add
 PUT /pet/setAdopter
 PUT /pet/prolongTrial14/{petId}
 PUT /pet/prolongTrial30/{petId}
 PUT /pet/cancelTrial/{petId}
 GET /dog-shelter/reports/{pet_id}
 GET /dog-shelter/overdue-reports/

vzavolsky committed:
Merge pull request #29 from java-dev-int-4-team-1/feature-V-20230711

Report API.

vzavolsky committed:
Merge pull request #30 from java-dev-int-4-team-1/feature-S-20230714-Dialog-Completing

Feature s 20230714 dialog completing

vzavolsky committed:
Контроллеры.

vzavolsky committed:
Merge branch 'dev' into feature-V-20230711

vzavolsky committed:
pet.setAdoptionDate(LocalDate.now().plus(30, ChronoUnit.DAYS));

vzavolsky committed:
pet.setAdoptionDate(LocalDate.now().plus(30, ChronoUnit.DAYS));
pet.setAdopter(null);

vzavolsky committed:
Merge pull request #32 from java-dev-int-4-team-1/feature-V-20230711

Контроллеры ч.2

sergei-gots committed:
CREATE TABLE order changed :  adopters before pet
CHANGED COLUMN TYPE reports.sentTime (TIMESTAMP) to reports.sent (DATE)
LocalDateTime Report.sentTime refactored --> LocalDate Report.sent

vzavolsky committed:
Чистка дублей классов + insert-info-pets.xml.

vzavolsky committed:
Чистка дублей классов + insert-info-pets.xml.

vzavolsky committed:
Чистка дублей классов + insert-info-pets.xml.

vzavolsky committed:
Чистка дублей классов + insert-info-pets.xml.

vzavolsky committed:
Update application.properties

spring.servlet.multipart.max-file-size = 5MB
spring.servlet.multipart.max-request-size = 5MB

vzavolsky committed:
Чистка дублей классов + insert-info-pets.xml.

vzavolsky committed:
Чистка дублей классов + insert-info-pets.xml.

vzavolsky committed:
Merge pull request #35 from java-dev-int-4-team-1/feature-V-20230711

Небольшие правки

vzavolsky committed:
AdopterController/Service
PetContrloller/Service
ShelterContorller/Service
VolunteerController/Service
ReportController/Service

vzavolsky committed:
Report - imgPath.

vzavolsky committed:
.

vzavolsky committed:
.

vzavolsky committed:
.

vzavolsky committed:
Merge pull request #36 from java-dev-int-4-team-1/feature-V-20230711

Контроллеры, сервисы.

sergei-gots committed:
Initial Commit: db.changeset insert-into-volunteers.xml changed

vzavolsky committed:
Отчёты (миграции), выборка животных для которых нет "вчерашнего" отчёта (работает относительно гринвича).

vzavolsky committed:
Добавлен shelterId для выборки отчётов по определённому приюту.

vzavolsky committed:
Добавлен shelterId для выборки отчётов по определённому приюту.

vzavolsky committed:
Добавлен shelterId для выборки отчётов по определённому приюту.

sergei-gots committed:
FIXED: Меню выбора шелтера убирается после выбора шелтера пользователем

vzavolsky committed:
Merge pull request #37 from java-dev-int-4-team-1/feature-V-20230711

Отчёты, выборка GET /reports/overdue.

vzavolsky committed:
Тестирование - начало.

sergei-gots committed:
REDUCED: Формирование меню пользователя

sergei-gots committed:
Adopter.chatMenuMessageId introduced

sergei-gots committed:
Test patched

sergei-gots committed:
Merge pull request #38 from java-dev-int-4-team-1/feature-S-20230716-dialog-completing

Feature s 20230716 dialog completing

vzavolsky committed:
Тестирование - начало.

sergei-gots committed:
Test ShelterInfoHandlerTest moved to package handler

sergei-gots committed:
Merge pull request #40 from java-dev-int-4-team-1/feature-S-20230716-dialog-completing

Test ShelterInfoHandlerTest moved to package handler

sergei-gots committed:
Merge branch 'dev' into feature-V-20230711

vzavolsky committed:
Merge pull request #39 from java-dev-int-4-team-1/feature-V-20230711

Формальный pull request, чтобы забрать обновления...

sergei-gots committed:
ShelterHandler removed. Выправлена работа по коррекции навигации внутри меню телеграм-бота

sergei-gots committed:
ShelterInfoHandler: Adopter.ChatShelter introduced

BizinMitya commented on src/main/java/pro/sky/petshelterbot/controllers/AdopterController.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    /* PUT /pet/setAdopter */
+    @PutMapping("/setAdopter/{petId}")
```
не нужен никакой setAdopter, просто /{petId}

dmitry-bizin commented on src/main/java/pro/sky/petshelterbot/controllers/AdopterController.java:
```diff
@@ -0,0 +1,57 @@
+    }
+
+    /* PUT /pet/cancelTrial/{petId} */
+    @PutMapping("cancelTrial/{petId}")
```
это скорее delete

dmitry-bizin commented on src/main/java/pro/sky/petshelterbot/controllers/PetController.java:
```diff
@@ -0,0 +1,81 @@
+            "Показывает сохраненные значения из БД и сообщает, " +
+            "что данные о животном сохранены или не сохранены.")
+    public ResponseEntity<Pet> add(
+            @RequestBody Pet pet,
```
так не будет работать - нужна аннотация RequestPart или RequestParam с указанием названия части запроса

vzavolsky replied:
Исправил. Но во время проверки изображение успешно добавлялось.

dmitry-bizin replied:
а как проверял? как запрос формировал?

vzavolsky replied:
Проверяю через Swagger.
![image](https://github.com/java-dev-int-4-team-1/pet-shelter-bot/assets/124736193/4026a4b0-4c34-4e9a-ac2f-9663c110f0f6)

dmitry-bizin replied:
Ну а pet-то где отправляешь? json? и через сваггер такие мультипартовые запросы не проверишь, он не умеет их делать - когда в одном запросе json и картинка, надо писать руками через постман или идеевский http

vzavolsky replied:
Убрал этот метод, чтобы никого не смущал.
Добавление картинки будет происходить отдельно от добавления животного.

dmitry-bizin commented on src/main/java/pro/sky/petshelterbot/controllers/VolunteerController.java:
```diff
@@ -0,0 +1,47 @@
+    POST /dog-shelter/volunteers/ */
+    @PostMapping
+    @ApiResponse(description = "Добавляет волонтёра в список волонтёров шелтера")
+    public ResponseEntity<Volunteer> AddVolunteerToShelter(@RequestBody Volunteer volunteer) {
```
названия методов с маленькой буквы

dmitry-bizin commented on src/main/java/pro/sky/petshelterbot/entity/Shelter.java:
```diff
@@ -0,0 +1,101 @@
+    private String tel;
+    private String email;
+
+    private String type;
```
enum

dmitry-bizin commented on src/main/java/pro/sky/petshelterbot/service/AdopterService.java:
```diff
@@ -0,0 +1,76 @@
+        return pet;
+    }
+
+    public Pet prolongTrial30(Long petId) {
```
хорошо бы убрать 30 и сделать параметром, при вызове передавать 30 - будет универсально

dmitry-bizin replied:
типа prolongTrialForNDays и передавать n

dmitry-bizin commented on src/main/java/pro/sky/petshelterbot/service/PetService.java:
```diff
@@ -0,0 +1,131 @@
+    }
+
+    /* POST /dog-shelter/volunteers/ */
+    public Volunteer AddVolunteerToShelter(Volunteer volunteer) {
```
названия методов с маленькой буквы

dmitry-bizin commented on src/main/java/pro/sky/petshelterbot/repository/ReportRepository.java:
```diff
@@ -0,0 +1,31 @@
+    @Query("from Report r where r.checked = null or r.checked = false and r.pet.shelter.id = :shelterId")
+    Page<Report> findAllUncheckedReports(Long shelterId, Pageable pageable);
+
+    @Query("select p from Pet p, Report r where p.adoptionDate is not null and p.adoptionDate >= current_timestamp and p = r.pet and p.shelter.id = :shelterId and not exists (select r from Report r where r.sent >= current_date - 1) group by p")
```
а почему full join? лучше бы сделать inner/left

vzavolsky committed:
Правки по комментариям от Дмитрия.

vzavolsky committed:
Merge pull request #42 from java-dev-int-4-team-1/feature-V-20230711

Правки по комментариям от Дмитрия.

sergei-gots committed:
Базовая навигация по меню бота completed

sergei-gots committed:
Merge branch 'dev' into feature-S-20230716-dialog-completing

sergei-gots committed:
Merge pull request #41 from java-dev-int-4-team-1/feature-S-20230716-dialog-completing

Feature s 20230716 dialog completing

sergei-gots committed:
Разрешение конфликтов с тестами

sergei-gots committed:
Merge pull request #43 from java-dev-int-4-team-1/feature-S-20230716-dialog-completing

Разрешение конфликтов с тестами

sergei-gots committed:
Диалог с волонтёром. Коррекция сценария. Подготовка.

sergei-gots committed:
Merge pull request #44 from java-dev-int-4-team-1/feature-S-20230716-dialog-completing

Feature s 20230716 dialog completing

sergei-gots committed:
Диалог с волонтёром. Введена отмена заявки на диалог.

sergei-gots committed:
Диалог с волонтёром. Отмена заявки на диалог completed

vzavolsky committed:
Правки по комментариям от Дмитрия.

vzavolsky committed:
Merge pull request #45 from java-dev-int-4-team-1/feature-V-20230711

Правки PetContoller

sergei-gots committed:
Диалог с волонтёром. COMPLETED

sergei-gots committed:
Merge pull request #46 from java-dev-int-4-team-1/feature-S-20230716-dialog-completing

Feature s 20230716 dialog completing

sergei-gots committed:
Диалог с волонтёром. Fix когда пользователь перевыбирает шетлер COMPLETED

sergei-gots committed:
Диалог с волонтёром. Сообщение об ограничении функциональности на фазе ожидания волонтёра

sergei-gots committed:
Navigation fixes

sergei-gots committed:
Merge pull request #47 from java-dev-int-4-team-1/feature-S-20230716-dialog-completing

Feature s 20230716 Navigation fixes

sergei-gots committed:
Navigation fixes-2. Completed

sergei-gots committed:
Merge pull request #48 from java-dev-int-4-team-1/feature-S-20230716-dialog-completing

Feature s 20230716 Navigation fixes . Completed

vzavolsky committed:
Тестирование - проблемы создания таблиц.

vzavolsky committed:
Application properties для тестов.

vzavolsky committed:
Тестирование.

sergei-gots committed:
Navigation Refactoring. Completed

sergei-gots committed:
Merge pull request #49 from java-dev-int-4-team-1/feature-S-20230716-dialog-completing

Navigation Refactoring. Completed

sergei-gots committed:
AdopterInputHandler introduced. Basic algorithm provided.

sergei-gots committed:
AdopterInputHandler introduced. Basic algorithm provided.

vzavolsky committed:
Тестирование.

sergei-gots committed:
ChatStat.INITIAL_STATE added

sergei-gots committed:
column phone_number inroduced

sergei-gots committed:
Input contacts completed

sergei-gots committed:
Merge pull request #50 from java-dev-int-4-team-1/feature-S-20230716-dialog-completing

Feature s 20230718 Adopter Input Algorithm introduced

sergei-gots committed:
TelegramBotListener and handlers  refactored

sergei-gots committed:
Merge pull request #52 from java-dev-int-4-team-1/feature-S-20230716-dialog-completing

TelegramBotListener and handlers  refactored

sergei-gots committed:
Navigation fixes

sergei-gots committed:
Merge pull request #53 from java-dev-int-4-team-1/feature-S-20230716-dialog-completing

Navigation fixes for /start and /reset

sergei-gots committed:
Stage create Report due Enter Report Data by Adopter added

sergei-gots committed:
Stage create Report due Enter Report Data by Adopter added

sergei-gots committed:
Input report's texts

sergei-gots committed:
Merge pull request #54 from java-dev-int-4-team-1/feature-S-20230716-dialog-completing

Report's texts input

sergei-gots committed:
Merge branch 'feature-V-20230711' into feature-S-20230716-dialog-completing

sergei-gots committed:
Merge remote-tracking branch 'origin/feature-S-20230716-dialog-completing' into feature-S-20230716-dialog-completing

# Conflicts:
#	src/main/java/pro/sky/petshelterbot/repository/PetRepository.java

sergei-gots committed:
Menu navigation fixes

sergei-gots committed:
Handler.handlePhoto introduced

sergei-gots committed:
Merge pull request #55 from java-dev-int-4-team-1/feature-S-20230716-dialog-completing

handlePhoto introduced

sergei-gots committed:
AdopterInputHandler.handlePhoto() is being develoved

sergei-gots committed:
AdopterInputHandler.handlePhoto() is being develoved

sergei-gots committed:
Merge pull request #56 from java-dev-int-4-team-1/feature-S-20230716-dialog-completing

AdopterInputHandler.handlePhoto

sergei-gots committed:
ReportRepository r.sent is not null

sergei-gots committed:
Merge pull request #57 from java-dev-int-4-team-1/feature-S-20230716-dialog-completing

ReportRepository Fix:  report.set is not null added

sergei-gots committed:
FileManager.saveTelegramFile added not checked yet

sergei-gots committed:
Update ReportRepository.java resolve conflicts

sergei-gots committed:
Add pet's photo to the report added

sergei-gots committed:
Merge pull request #59 from java-dev-int-4-team-1/feature-S-20230716-dialog-completing

Pet Photo

sergei-gots committed:
Bot Reports-part completed

sergei-gots committed:
Merge pull request #60 from java-dev-int-4-team-1/feature-S-20230716-dialog-completing

Bot Report Upload part completed

sergei-gots committed:
Notifier added

sergei-gots committed:
Dialog fixed

sergei-gots committed:
Merge pull request #61 from java-dev-int-4-team-1/feature-S-20230716-dialog-completing

Bot completed

sergei-gots committed:
Initial commit

sergei-gots committed:
ShelterServiceTest.add -test is being prepared

sergei-gots committed:
test/resources/application.properties db.img.path added

sergei-gots committed:
test/resources/application.properties NON_KEYWORDS=KEY added to spring.datasource.url

sergei-gots committed:
Merge pull request #62 from java-dev-int-4-team-1/feature-S-API-ShelterService-test

Test issues resolved.

sergei-gots committed:
1st working test

sergei-gots committed:
FIRST Integration Test added: ShelterControllerTest.add

sergei-gots committed:
FIRST Integration Test added: ShelterControllerTest.add

sergei-gots committed:
FIRST Unit Test added: ShelterServiceTest.add_get_findAll

sergei-gots committed:
Unit Test tried to be added: ShelterControllerTest.add-case

sergei-gots committed:
Unit Test tried to be added: ShelterControllerTest.add-case

sergei-gots committed:
Unit Test tried to be added: ShelterControllerTest.add-case

sergei-gots committed:
AppTest contectLoads added

sergei-gots committed:
ShelterControllerTest.add ✓

sergei-gots committed:
ShelterControllerTest ✓

sergei-gots committed:
i. Hanlder refactored: split : ac. UpdateHandler introduced

sergei-gots committed:
i. TelegramBotListener refactored: split : c. DefaultHandler introduced

sergei-gots committed:
Tests for ShelterService/Controller completed

sergei-gots committed:
Merge pull request #63 from java-dev-int-4-team-1/feature-S-API-ShelterService-test

Shelter: Controller/Service-tests completed

sergei-gots committed:
TelegramBotListenerTest initial commit

sergei-gots committed:
TelegramBotListenerTest update-case-dependent text change fixed

sergei-gots committed:
TelegramBotListenerTest next step try: research in providing process(Updates) based on Hanler.handle(update) calls

sergei-gots committed:
TestAdopterController introduced

sergei-gots committed:
Merge pull request #64 from java-dev-int-4-team-1/feature-S-tests

AdopterControllerTest introduced

sergei-gots committed:
TestAdopterController completed

sergei-gots committed:
PetControllerTest added

sergei-gots committed:
VolunteerController/Service fixed; VolunteerControllerTest completed

sergei-gots committed:
PetController fixed

sergei-gots committed:
ReportControllerTest 3/7 covered

sergei-gots committed:
Contoller's Unit Tests completed

sergei-gots committed:
VolunteerService Unit Tests added

sergei-gots committed:
AdopterService Unit Tests added

sergei-gots committed:
PetService refactored

sergei-gots committed:
.handler.* refactored

sergei-gots committed:
ReportServiceTest added

sergei-gots committed:
Bot menu navigation fixed

sergei-gots committed:
Swagger-Doc ShelterController.findAll added

sergei-gots committed:
Swagger-Doc PetController.findByShelterId added

sergei-gots committed:
Swagger-Doc for ReportController.approve/disapprove added

sergei-gots committed:
Swagger-Doc for ReportController.approve/disapprove added

sergei-gots committed:
Swagger-Doc for AdopterController

sergei-gots committed:
resources/swagger-doc/adopter.json, pet.json added

sergei-gots committed:
PetControllerTest fixed

sergei-gots committed:
All Necessary For Presentation swagger-ui doc added not fixed

sergei-gots committed:
Swagger-doc: @Content(schema-definition) fixed

sergei-gots committed:
Swagger-doc: @Operation introduced

sergei-gots committed:
TelegramBotListener split: handler.Handlers added

sergei-gots committed:
API doc for swagger - works terminated and finished as is.

sergei-gots committed:
Merge pull request #65 from java-dev-int-4-team-1/feature-S-tests

unit tests

sergei-gots committed:
application-prod.properties added

sergei-gots committed:
bot fixes

sergei-gots committed:
bot refactored

sergei-gots committed:
ListenerTest basic fixes

sergei-gots committed:
ListenerTest basic fixes: tex_update.json fixed

sergei-gots committed:
Tests: text_update.json fixed

sergei-gots committed:
TelegramBotListenerTest : /start example completed

sergei-gots committed:
Merge pull request #66 from java-dev-int-4-team-1/feature-S-tests

TelegramBotListenerTest : /start example completed

sergei-gots committed:
pom.xml reduced

sergei-gots committed:
Merge pull request #67 from java-dev-int-4-team-1/feature-S-tests

pom.xml reduced
